### DISCUSSION TITLE
Bypass Row Level Security.

### DISCUSSION THREAD

**Discussion by EdgarsLv** [supabase/supabase#13068] [Questions] [answered]
Wasn't createClient with secret key suppose to bypass row level serurity?

Or has something changed? And how then to bypass row level security?

1st query with rules on - 3 results.
2nd query with rules off - 365 results.

For select simple rules: auth id = user id

![image](https://user-images.githubusercontent.com/56837461/226002899-e627c801-1d99-424c-832b-f514b73af62d.png)

**Comment by GaryAustin1** [comment, marked as answer]
Service role bypasses RLS.   So either you don’t have a service role key, or you are using a service role key and then setting a user session, or using auth-helpers which sets a user session that will put the user jwt in the authorization header.

**Reply by EdgarsLv** [reply]
Yes that was because of user session.

**Comment by EdgarsLv** [comment]
And one more question.

Is there way to fetch data from auth.users?

**Comment by GaryAustin1** [comment]
No, you have use a profile table or an rpc security definer function. Sent from my iPhoneOn Mar 17, 2023, at 2:42 PM, EdgarsLv ***@***.***> wrote:﻿
And one more question.
Is there way to fetch data from auth.users?
